### DISCUSSION TITLE
Installation - out of memory when building js part

### DISCUSSION THREAD

**Discussion by tmarly** [familytree365/genealogy#1668] [Q&A] [answered]
Hi,

I cloned the master branch (commit 609faa5) and followed install instructions, however I'm having troubles with this part:
```bash
cd client
yarn
yarn build
```
=> It took a long time (may be 10 minutes), before displaying:
```
FATAL ERROR: Ineffective mark-compacts near heap limit Allocation failed - JavaScript heap out of memory
 1: 0x9feb00 node::Abort() [/usr/bin/node]
 2: 0x94a471 node::FatalError(char const*, char const*) [/usr/bin/node]
 3: 0xb720fe v8::Utils::ReportOOMFailure(v8::internal::Isolate*, char const*, bool) [/usr/bin/node]
 4: 0xb72477 v8::internal::V8::FatalProcessOutOfMemory(v8::internal::Isolate*, char const*, bool) [/usr/bin/node]
 5: 0xd1c065  [/usr/bin/node]
 6: 0xd1cbef  [/usr/bin/node]
 7: 0xd2ac5b v8::internal::Heap::CollectGarbage(v8::internal::AllocationSpace, v8::internal::GarbageCollectionReason, v8::GCCallbackFlags) [/usr/bin/node]
 8: 0xd2d315 v8::internal::Heap::HandleGCRequest() [/usr/bin/node]
 9: 0xcd3977 v8::internal::StackGuard::HandleInterrupts() [/usr/bin/node]
10: 0x104200a v8::internal::Runtime_StackGuardWithGap(int, unsigned long*, v8::internal::Isolate*) [/usr/bin/node]
11: 0x13ce059  [/usr/bin/node]
/bin/sh: line 1: 27486 Aborted                 (core dumped) vue-cli-service build --no-clean
```

I also tried with:
```bash
npm install
npm run build
```

But same result.

I tried to increase memory with:
```bash
sudo npm install -g increase-memory-limit
increase-memory-limit
npm run build
```
But the command took ages, I had to kill it.

My environment:
- Ubuntu 18.04
- Node 14.8.0
- Yarn 1.22.4
- Npm 6.14.7
- RAM: 8Go

Any clue ?

Thanks

**Comment by curtisdelicata** [comment]
Please see familytree365/backend and familytree365/frontend we abandoned the Laravel Enso version.

**Comment by tmarly** [comment]
ok thanks Curtis

**Comment by curtisdelicata** [comment, marked as answer]
Now working please see this repository and nuxt.

**Comment by alcyoneus-g** [comment]
@everyone you got scammed, please check @curtisdelicata 's Linkedin profile. He is fraud and liar. He never pay for developers. 

He runs fake business called FamilyTree365. He lies developers to contribute the project for a week and pretend to pay after a week but he disappears after receiving PR.

I should alert other honest developers and customers that you should never support Curtis Delicata since he is a fraud.

I can share you more evidence if anybody interested.

https://www.linkedin.com/feed/update/urn:li:activity:6906662303572606976/

![image](https://user-images.githubusercontent.com/72995735/157096643-3d6dd048-5dcf-47ca-b5c4-f97d4064134b.png)

**Comment by alcyoneus-g** [comment]
> Hi,
> 
> I cloned the master branch (commit [609faa5](https://github.com/familytree365/genealogy/commit/609faa5df3bb40c8afc4978a0ff95341e27c58e7)) and followed install instructions, however I'm having troubles with this part:
> 
> ```shell
> cd client
> yarn
> yarn build
> ```
> 
> => It took a long time (may be 10 minutes), before displaying:
> 
> ```
> FATAL ERROR: Ineffective mark-compacts near heap limit Allocation failed - JavaScript heap out of memory
>  1: 0x9feb00 node::Abort() [/usr/bin/node]
>  2: 0x94a471 node::FatalError(char const*, char const*) [/usr/bin/node]
>  3: 0xb720fe v8::Utils::ReportOOMFailure(v8::internal::Isolate*, char const*, bool) [/usr/bin/node]
>  4: 0xb72477 v8::internal::V8::FatalProcessOutOfMemory(v8::internal::Isolate*, char const*, bool) [/usr/bin/node]
>  5: 0xd1c065  [/usr/bin/node]
>  6: 0xd1cbef  [/usr/bin/node]
>  7: 0xd2ac5b v8::internal::Heap::CollectGarbage(v8::internal::AllocationSpace, v8::internal::GarbageCollectionReason, v8::GCCallbackFlags) [/usr/bin/node]
>  8: 0xd2d315 v8::internal::Heap::HandleGCRequest() [/usr/bin/node]
>  9: 0xcd3977 v8::internal::StackGuard::HandleInterrupts() [/usr/bin/node]
> 10: 0x104200a v8::internal::Runtime_StackGuardWithGap(int, unsigned long*, v8::internal::Isolate*) [/usr/bin/node]
> 11: 0x13ce059  [/usr/bin/node]
> /bin/sh: line 1: 27486 Aborted                 (core dumped) vue-cli-service build --no-clean
> ```
> 
> I also tried with:
> 
> ```shell
> npm install
> npm run build
> ```
> 
> But same result.
> 
> I tried to increase memory with:
> 
> ```shell
> sudo npm install -g increase-memory-limit
> increase-memory-limit
> npm run build
> ```
> 
> But the command took ages, I had to kill it.
> 
> My environment:
> 
> * Ubuntu 18.04
> * Node 14.8.0
> * Yarn 1.22.4
> * Npm 6.14.7
> * RAM: 8Go
> 
> Any clue ?
> 
> Thanks

@everyone you got scammed, please check @curtisdelicata 's Linkedin profile. He is fraud and liar. He never pay for developers. 

He runs fake business called FamilyTree365. He lies developers to contribute the project for a week and pretend to pay after a week but he disappears after receiving PR.

I should alert other honest developers and customers that you should never support Curtis Delicata since he is a fraud.

I can share you more evidence if anybody interested.

https://www.linkedin.com/feed/update/urn:li:activity:6906662303572606976/

![image](https://user-images.githubusercontent.com/72995735/157096643-3d6dd048-5dcf-47ca-b5c4-f97d4064134b.png)
